src/test/java/com/wikia/webdriver/testcases/messagewall/MessageWallTests.java

```diff
@@ -1,64 +1,83 @@
 package com.wikia.webdriver.testcases.messagewall;
 
+import com.wikia.webdriver.common.core.TestContext;
 import com.wikia.webdriver.common.core.annotations.RelatedIssue;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.special.SpecialWikiActivityPageObject;
+
+import org.openqa.selenium.TimeoutException;
 import org.openqa.selenium.interactions.Actions;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 import com.wikia.webdriver.common.contentpatterns.PageContent;
 import com.wikia.webdriver.common.contentpatterns.SourceModeContent;
 import com.wikia.webdriver.common.core.annotations.Execute;
-import com.wikia.webdriver.common.core.configuration.Configuration;
 import com.wikia.webdriver.common.core.helpers.User;
-import com.wikia.webdriver.common.properties.Credentials;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.pageobjectsfactory.componentobject.minieditor.MiniEditorComponentObject;
 import com.wikia.webdriver.pageobjectsfactory.componentobject.minieditor.MiniEditorPreviewComponentObject;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.messagewall.MessageWall;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.messagewall.MessageWallCloseRemoveThreadPageObject;
-import com.wikia.webdriver.pageobjectsfactory.pageobject.special.block.SpecialBlockListPage;
-import com.wikia.webdriver.pageobjectsfactory.pageobject.special.block.SpecialBlockPage;
+
+import java.lang.reflect.Method;
 
 public class MessageWallTests extends NewTestTemplate {
 
-  Credentials credentials = Configuration.getCredentials();
+  private MessageWall wall;
+
+  private String userName;
+
+  /**
+   * Setup test environment - navigate to Message Wall page of user
+   */
+  @BeforeMethod(alwaysRun = true)
+  public void loadWallPageAndCheckLoginStatus(Method method) {
+    User user = method.getAnnotation(Execute.class).asUser();
+
+    userName = user.getUserName();
+    wall = new MessageWall(driver).open(userName);
+
+    // SUS-1737: In Wall tests sometimes user is not authenticated properly
+    // Let's check for this, and if this happens, log the situation and log in again
+    try {
+      wall.verifyUserLoggedIn(user);
+    } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)